rllib/algorithms/appo/tests/test_appo.py

```diff
@@ -117,30 +117,30 @@ def test_appo_entropy_coeff_schedule(self):
         )
 
         def _step_n_times(algo, n: int):
-            """Step Algorithm n times.
-
-            Returns:
-                learning rate at the end of the execution.
-            """
             for _ in range(n):
                 results = algo.train()
                 print(algo.workers.local_worker().global_vars)
                 print(results)
-            return results["info"][LEARNER_INFO][DEFAULT_POLICY_ID][LEARNER_STATS_KEY][
-                "entropy_coeff"
-            ]
+            return (
+                results["info"][LEARNER_INFO][DEFAULT_POLICY_ID][LEARNER_STATS_KEY][
+                    "entropy_coeff"
+                ],
+                results["num_env_steps_sampled"],
+            )
 
         for _ in framework_iterator(config):
             algo = config.build(env="CartPole-v1")
 
-            coeff = _step_n_times(algo, 10)  # 200 timesteps
-            # Should be close to the starting coeff of 0.01.
-            self.assertLessEqual(coeff, 0.01)
-            self.assertGreaterEqual(coeff, 0.001)
+            coeff, num_env_steps_sampled = _step_n_times(algo, 5)  # ~100 timesteps
+            if num_env_steps_sampled > 300:
+                self.assertLessEqual(coeff, 0.001)
+                self.assertGreaterEqual(coeff, 0.0001)
+            else:
+                self.assertLessEqual(coeff, 0.01)
+                self.assertGreaterEqual(coeff, 0.001)
```

Comment:
Can we redesign this test a little bit?

It could be simpler for example by doing:
```
entropy_coeff_schedule=[[0, 0.1], [200, 0.001], [600, 0.0001]]
```

Also,  `_step_n_times()` should be "step_until_n_steps_reached()".
We should then be able to reuse this with entropy coefficient tests for other algorithms if so desired.
The "~100 timesteps" thing can easily change per algorithm or when something else in the algorithm under test changes that has nothing to do with the coefficient schedule.